packages/react/src/components/pickers/BasePicker.tsx

```diff
@@ -183,6 +188,10 @@ export class BasePicker<T extends {}, P extends IBasePickerProps<T>>
       }
     }
 
+    if (!!oldProps.required !== !!this.props.required) {
```

Comment:
I think the check would need to be whether `onGetErrorMessage` returns a string, since there could be other types of validation based on selected items & value beyond the required prop.